src/dom/document.ts

```diff
@@ -307,11 +307,11 @@ export class Document extends Node {
     return doc;
   }
 
-  querySelector(selectors: string): Element | null {
-    return this._nwapi.first(selectors, this);
+  querySelector<T = Element>(selectors: string): T | null {
```

Comment:
I think that `T extends Element` is more preferable tbh, afaik `querySelector` can't return anything that's not an element

Reply:
I used `T extends Element` at first, but I think it is better for users to leave it this way (at least until deno-dom implements all the `HTML*Element`). 

Not enforcing this constraint make it compatible with `<reference lib="dom" />`, so consumers are able to have better typings, for example:
```ts
document.querySelectorAll<HTMLInputElement>("input").forEach(input => input.value)
```
I understand that behaviours may not be actually implemented so casting could be dangerous, but I feel like for now it's still better to delegate the decision to the user because it allows them to write isomorphic code that is compatible for both virtual/actual dom

Reply:
That's a good point, yeah. Okay, this is fine, np.